### PR TITLE
Restrict uses of the client and sendMessage to the protocol directory

### PR DESCRIPTION
**Pull request by jasonLaster**
This PR restricts uses of protocol/socket's client and sendMessage to the protocol directory. This is a nice win because it reduces the number of ways in which the application can talk to the protocol.

NOTE: this would not have been possible a couple of months ago before Brian and Mark started refactoring our protocol

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -42,6 +42,11 @@
             "name": "react-redux",
             "importNames": ["useDispatch", "useSelector"],
             "message": "Please import 'useAppDispatch/useAppSelector' from 'ui/setup/hooks' instead."
+          },
+          {
+            "name": "protocol/socket",
+            "importNames": ["client", "sendMessage"],
+            "message": "Please use ThreadFront."
```
Comment (bvaughn) [inline .eslintrc.json:49]: 😍

File: packages/protocol/thread/thread.ts [hunk 2]
```diff
@@ -883,6 +899,23 @@ class _ThreadFront {
...
   }
 
+  findMessagesInRange(range: PointRange) {
+    return client.Console.findMessagesInRange({ range }, ThreadFront.sessionId!);
+  }
+
+  async findKeyboardEvents(onKeyboardEvents: (events: keyboardEvents) => void) {
+    const keyboardEventsLoaded = client.Session.findKeyboardEvents({}, this.sessionId!);
+    client.Session.addKeyboardEventsListener(onKeyboardEvents);
+    return keyboardEventsLoaded
```
Comment (bvaughn) [inline packages/protocol/thread/thread.ts:909]: The order of these two should probably be reversed (add the handler before calling `findKeyboardEvents`)
```suggestion
    client.Session.addKeyboardEventsListener(onKeyboardEvents);
    return client.Session.findKeyboardEvents({}, this.sessionId!);
```
Might not matter in this case but I think it's a good general practice to follow (so you don't miss any potentially synchronous events).

File: packages/protocol/thread/thread.ts [hunk 3]
```diff
@@ -883,6 +899,23 @@ class _ThreadFront {
...
+  }
+
+  async findNavigationEvents(onNavigationEvents: (events: navigationEvents) => void) {
+    const navigationEventsLoaded = client.Session.findNavigationEvents({}, this.sessionId!);
+    client.Session.addNavigationEventsListener(onNavigationEvents);
+    return navigationEventsLoaded
```
Comment (bvaughn) [inline packages/protocol/thread/thread.ts:915]: Same as above (add the handler before calling `findNavigationEvents`)
```suggestion
    client.Session.addNavigationEventsListener(onNavigationEvents);
    return client.Session.findNavigationEvents({}, this.sessionId!);
```

File: packages/protocol/thread/thread.ts (hunk 2 above)
Comment (bvaughn) [reply packages/protocol/thread/thread.ts:909]: PS Surprised the linter didn't complain about the missing semicolon on line 909

File: pages/recording/[id]/sourcemap/[sourceId].tsx [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import { useRouter } from "next/router";
 import React, { useEffect, useState } from "react";
+// eslint-disable-next-line no-restricted-imports
 import { initSocket, client } from "protocol/socket";
```
Comment (bvaughn) [inline pages/recording/[id]/sourcemap/[sourceId].tsx:4]: This use is still okay?

File: src/ui/actions/app.ts [hunk 5]
```diff
@@ -161,8 +160,8 @@ function onKeyboardEvents(events: KeyboardEvent[], store: UIStore) {
   });
 }
 
-function onNavigationEvents(events: ReplayNavigationEvent[], store: UIStore) {
-  const currentNavEvents: ReplayNavigationEvent[] = events.map(e => ({ ...e, kind: "navigation" }));
+function onNavigationEvents(events: navigationEvents, store: UIStore) {
```
Comment (bvaughn) [inline src/ui/actions/app.ts:163]: Oh that was bizarre before.

**Review by bvaughn — APPROVED** [review]
Nice! Couple of small nits.

File: pages/recording/[id]/sourcemap/[sourceId].tsx (hunk 4 above)
Comment (jasonLaster) [reply pages/recording/[id]/sourcemap/[sourceId].tsx:4]: The sourcemap visualizer is its own beast.

File: src/ui/actions/app.ts (hunk 5 above)
Comment (jasonLaster) [reply src/ui/actions/app.ts:163]: a tad

File: packages/protocol/thread/thread.ts (hunk 3 above)
Comment (jasonLaster) [reply packages/protocol/thread/thread.ts]: good call

File: .eslintrc.json (hunk 1 above)
Comment (JessicaSachs) [reply .eslintrc.json:49]: Yeah, okay, that's nice. Def stealing this as a tip.

**Pull request merged**